x-pack/auditbeat/module/system/socket/guess/guess.go

```diff
@@ -233,11 +232,14 @@ func guessOnce(guesser Guesser, installer helper.ProbeInstaller, ctx Context) (r
 
 		case err := <-perfchan.ErrC():
 			if err != nil {
-				return nil, errors.Wrap(err, "error received from perf channel")
+				return nil, fmt.Errorf("error received from perf channel: %w", err)
 			}
 
 		case <-perfchan.LostC():
-			return nil, errors.Wrap(err, "event loss in perf channel")
+			if err != nil {
```

Comment:
Which `err` is this checking? I wasn't expecting this check.

Reply:
Yeah 🙂. There is no error; this check retains the exact semantics of the original code since the `Wrap` call will always see a nil error and so return nil. Really this should return `nil, nil`. I'm happy to make that change and really only left this here to avoid a concern about the loss of the return.

Reply:
Ok, I understand now. Thanks for the explanation. Let's remove it.

Reply:
Done